src/libstd/sys/unix/fd.rs

```diff
@@ -243,6 +243,30 @@ impl FileDesc {
         }
         cvt(unsafe { libc::fcntl(fd, libc::F_DUPFD, 0) }).and_then(make_filedesc)
     }
+
+    /// Replace the open file referred to by `self` with the open file
+    /// referred to by `other`.  This does not change the file
+    /// descriptor _number_ within self; rather, it changes what open
+    /// file that number refers to.
+    ///
+    /// On success, the file formerly referred to by `self` is
+    /// transferred to a newly-allocated file descriptor and returned.
+    /// On failure, the file referred to by `self` is unchanged.
+    /// Regardless of success or failure, `other` is consumed, which
+    /// means the file descriptor formerly held by `other` will be
+    /// closed.
+    ///
+    /// The file underlying `self` is replaced atomically, but the
+    /// _overall_ operation is not atomic; concurrent threads that snoop on
+    /// the set of valid file descriptors (which they shouldn't) can observe
+    /// intermediate states.
+    pub fn replace(&mut self, other: FileDesc) -> io::Result<FileDesc> {
```

Comment:
I don't think the replacement semantics are actually needed when closing, right? That is, I think it's safe for this to return `()` and avoid the first call to `duplicate` for the specific use case of `Stdout::close`?

Reply:
Yes, I realized the next morning that any caller that actually needs to preserve the old file can just call `duplicate` itself.  And that will dovetail better with the semantics of `SetStdHandle` on Windows.